src/console/clients/checker/checks/http.rs

```diff
@@ -1,97 +1,92 @@
-
-    for http_tracker in http_trackers {
-        let mut http_checker = CheckerOutput {
-            url: http_tracker.to_string(),
-            status: Status {
-                code: String::new(),
-                message: String::new(),
-            },
+#[derive(Debug, Clone, Serialize)]
+pub struct Checks {
+    url: Url,
+    results: Vec<(Check, Result<(), Error>)>,
+}
+
+#[derive(Debug, Clone, Serialize)]
+pub enum Check {
+    Announce,
+    Scrape,
+}
+
+pub async fn run(http_trackers: Vec<Url>, timeout: Duration) -> Vec<Result<Checks, Checks>> {
+    let mut results = Vec::default();
+
+    tracing::debug!("HTTP trackers ...");
+
+    for ref url in http_trackers {
+        let mut checks = Checks {
+            url: url.clone(),
+            results: Vec::default(),
         };
 
-        match check_http_announce(http_tracker).await {
-            Ok(()) => {
-                check_results.push(Ok(()));
-                http_checker.status.code = "ok".to_string();
-            }
-            Err(err) => {
-                check_results.push(Err(err));
-                http_checker.status.code = "error".to_string();
-                http_checker.status.message = "Announce is failing.".to_string();
-            }
+        // Announce
+        {
+            let check = check_http_announce(url, timeout).await.map(|_| ());
+
+            checks.results.push((Check::Announce, check));
         }
 
-        match check_http_scrape(http_tracker).await {
-            Ok(()) => {
-                check_results.push(Ok(()));
-                http_checker.status.code = "ok".to_string();
-            }
-            Err(err) => {
-                check_results.push(Err(err));
-                http_checker.status.code = "error".to_string();
-                http_checker.status.message = "Scrape is failing.".to_string();
-            }
+        // Scrape
+        {
+            let check = check_http_scrape(url, timeout).await.map(|_| ());
+
+            checks.results.push((Check::Scrape, check));
+        }
+
+        if checks.results.iter().any(|f| f.1.is_err()) {
+            results.push(Err(checks));
+        } else {
+            results.push(Ok(checks));
         }
-        http_checkers.push(http_checker);
     }
-    http_checkers
+
+    results
 }
 
-async fn check_http_announce(tracker_url: &Url) -> Result<(), CheckError> {
+async fn check_http_announce(url: &Url, timeout: Duration) -> Result<Announce, Error> {
     let info_hash_str = "9c38422213e30bff212b30c360d26f9a02136422".to_string(); // # DevSkim: ignore DS173237
     let info_hash = InfoHash::from_str(&info_hash_str).expect("a valid info-hash is required");
 
     // todo: HTTP request could panic.For example, if the server is not accessible.
     // We should change the client to catch that error and return a `CheckError`.
     // Otherwise the checking process will stop. The idea is to process all checks
     // and return a final report.
```

Comment:
Hi @da2ce7 this comment could be obsolete, I think the Http client does not pacnis anymore. It always return a Result.